lib/cassandra/tasks/seedregistry.rb

```diff
@@ -0,0 +1,161 @@
+require 'daemon_runner/shell_out'
+require 'daemon_runner/semaphore'
+
+module Cassandra
+  module Tasks
+   class SeedRegistry
+     # Create a new SeedRegistry task
+     #
+     # @param cluster_name [String] unique name (in Consul) for the Cassandra cluster
+     #
+     # @return [SeedRegistry]
+     #
+     def initialize cluster_name
+       @cluster_name = cluster_name.to_s
+       raise ArgumentError.new('cluster_name must not be empty') if @cluster_name.empty?
+       @semaphore = nil
+       @renew_thread = nil
+     end
+
+     # Schedule the sedd registration process to run periodically
+     #
+     def schedule
+       [:interval, '10m']
```

Comment:
I think there may also be some checks that the Consul agent is running to detect listener states of the Gossip, Thrift, and Native transports, if running `nodetool info` every 10s is prohibitive. Similarly, the DC and Rack values can probably be cached as they shouldn't change once the Cassandra node is in a healthy state